### PR TITLE
Add initial PR template

### LINKED ISSUE #7: Add pull request template
We should add a pull request template guiding contributors through opening a well-formed PR.

### PR DESCRIPTION
**Pull request by tsmaeder**
## What it does

Adds a pull request template (#7)

## How to test.
You can't really before merging. However, I would welcome a lively discussion on what should be in the template.

### CODE REVIEW

File: pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+<!-- Include relevant issues and describe how they are addressed. -->
+
+## How to test
+<!-- Explain how a reviewer can reproduce a bug, test new functionality or verify performance improvements. -->
+
+## Author checklist
+
+- I have thoroughly tested my changes
+- The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
+- I have signed the the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)
```
Comment (jjohnstn) [inline pull_request_template.md:21]: Minor typo (too many "the"s).  Is the ECA not enforced by github for Eclipse in the checks?

**Comment by tsmaeder** [comment]
@akurtakov

**Comment by gayanper** [comment]
Just to have a understanding, Are we making having a github issue optional here ?

File: pull_request_template.md (hunk 1 above)
Comment (jjohnstn) [reply pull_request_template.md]: Fixed.

**Review by sravanlakkimsetti — APPROVED** [review]

**Pull request merged**